src/main/scala/li/cil/oc/integration/forestry/UpgradeBeekeeperUtil.java

```diff
@@ -0,0 +1,202 @@
+package li.cil.oc.integration.forestry;
+
+import forestry.api.apiculture.IBeeHousing;
+import gregtech.api.metatileentity.BaseMetaTileEntity;
+import gregtech.api.util.GT_ApiaryUpgrade;
+import gregtech.api.util.GT_Utility;
+import gregtech.common.tileentities.machines.basic.GT_MetaTileEntity_IndustrialApiary;
+import li.cil.oc.util.BlockPosition;
+import net.minecraft.inventory.IInventory;
+import net.minecraft.item.ItemStack;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.world.World;
+
+/**
+ * Wrapper class for UpgradeBeekeeper item. Scala doesn't seem to like GT machines,
+ * so we wrap most functionality in a static java class
+ * */
+public final class UpgradeBeekeeperUtil {
+
+    // First upgrade slot index of Industrial Apiaries because it's private for some reason
+    private static final int UPGRADE_INDEX = 7;
```

Comment:
Instead of storing *implementation details* of Industrial Apiary, it's better to create necessary method at GT side.